public/js/main.js

```diff
@@ -50,59 +36,16 @@ window.actions = {
   selectSourceURL: actions.selectSourceURL
 };
 
-function renderRoot(component) {
-  const mount = document.querySelector("#mount");
-
-  // bail in test environments that do not have a mount
-  if (!mount) {
-    return;
-  }
-
-  ReactDOM.render(
-    React.createElement(
-      Provider,
-      { store },
-      dom.div({
-        className: "theme-light",
-        style: { flex: 1 }
-      }, React.createElement(component))
-    ),
-    mount
-  );
-}
+const sourceMap = require("./utils/source-map");
 
 function unmountRoot() {
   const mount = document.querySelector("#mount");
   ReactDOM.unmountComponentAtNode(mount);
 }
 
-function getTargetFromQuery() {
-  const href = window.location.href;
-  const nodeMatch = href.match(/ws=([^&#]*)/);
-  const firefoxMatch = href.match(/firefox-tab=([^&#]*)/);
-  const chromeMatch = href.match(/chrome-tab=([^&#]*)/);
-
-  if (nodeMatch) {
-    return { type: "node", param: nodeMatch[1] };
-  } else if (firefoxMatch) {
-    return { type: "firefox", param: firefoxMatch[1] };
-  } else if (chromeMatch) {
-    return { type: "chrome", param: chromeMatch[1] };
-  }
-
-  return null;
-}
-
-const connTarget = getTargetFromQuery();
-if (connTarget) {
-  startDebugging(connTarget, actions).then((tabs) => {
-    actions.newTabs(tabs);
-    actions.selectTab({ id: connTarget.param });
-    renderRoot(App);
-  });
+if (isDevelopment()) {
+  bootstrap(React, ReactDOM, App, actions, store);
```

Comment:
This is where I'd probably be a little frustrated if I was trying to learn this project; compare the above code to this and it's a lot less clear what's going on. There is a place for abstractions though, so I could be wrong. I just feel like having code that reads like "if they need to select a target, render the toolbox, otherwise we have a target so connect it and render our app" would be nice.